internal/component/common/loki/client/batch.go

```diff
@@ -20,6 +20,14 @@ const (
 	errMaxStreamsLimitExceeded = "streams limit exceeded, streams: %d exceeds limit: %d, stream: '%s'"
 )
 
+func isErrMaxStreamsLimitExceeded(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	return strings.HasPrefix(err.Error(), "streams limit exceeded")
```

Comment:
Instead of `errMaxStreamsLimitExceeded` being a string and us having to check a string prefix we should turn that into a proper error `errors.New("treams limit exceeded")`. Then in every place we return it we change to 
`fmt.Erorrf("%w, streams: %d exceeds limit: %d, stream: '%s'", errMaxStreamsLimitExceeded, streams, maxStream, lables)`.

Then the check for this error is just `errors.Is(err, errMaxStreamsLimitExceeded)`